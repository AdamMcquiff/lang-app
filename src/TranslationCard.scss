.m-category {
  text-align: center;
}