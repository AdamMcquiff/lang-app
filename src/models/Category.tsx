interface ICategoryModel {
    id?: number,
    title: string
}

export default ICategoryModel;