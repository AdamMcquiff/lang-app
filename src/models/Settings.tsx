export interface ISettingsModel {
    native_lang: string,
    translated_lang: string,
}