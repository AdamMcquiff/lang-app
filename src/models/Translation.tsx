interface ITranslationModel {
    id: number,
    native_word: string,
    translated_word: string,
    categoryId: number
}

export default ITranslationModel;