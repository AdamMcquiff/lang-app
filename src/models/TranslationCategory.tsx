interface ITranslationCategoryModel {
    id?: number,
    title: string
}

export default ITranslationCategoryModel;