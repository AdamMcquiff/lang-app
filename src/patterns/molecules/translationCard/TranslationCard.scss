.m-translation-card {
  text-align: center;
}