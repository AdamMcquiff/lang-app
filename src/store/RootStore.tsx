import SettingsService from '../services/settingsService/SettingsService';
import TranslationCategoryService from '../services/translationCategoriesService/TranslationCategoryService';
import TranslationService from '../services/translationService/TranslationService';
import SettingsStore from './SettingsStore';
import TranslationCategoryStore from './TranslationCategoryStore';
import TranslationStore from './TranslationStore';

class RootStore {
	public settingsStore: SettingsStore;
	public translationCategoryStore: TranslationCategoryStore;
	public translationStore: TranslationStore;

	constructor() {
		this.settingsStore = new SettingsStore(this, SettingsService);
		this.translationCategoryStore = new TranslationCategoryStore(this, TranslationCategoryService);
		this.translationStore = new TranslationStore(this, TranslationService);
	}
}

export default RootStore;