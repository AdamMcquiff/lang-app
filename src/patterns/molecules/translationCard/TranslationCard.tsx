import { observer } from 'mobx-react';
import * as React from 'react';
import './TranslationCard.scss';

interface ITranslationCard {
  translation: any,
}

@observer
class TranslationCard extends React.Component<ITranslationCard> {
  public render() {
    return (
      <div>
        {this.props.translation.native_word}
        {this.props.translation.translated_word}
      </div>
    );
  }
}

export default TranslationCard;
