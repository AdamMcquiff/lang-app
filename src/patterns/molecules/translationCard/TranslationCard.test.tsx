import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { ITranslationModel } from '../../../models/Translation';
import { RootStore } from '../../../store/RootStore';
import { TranslationCard } from './TranslationCard';

const rootStore = new RootStore();

describe('TranslationCard', () => {
    const translation: ITranslationModel = {
        categoryId: 1,
        native_word: "Hello",
        translated_word: "Hallo"
    };

    const props = {
        settingsStore: rootStore.settingsStore,
        translation,
        translationStore: rootStore.translationStore,
    }

    it('renders without crashing with all props', () => {
        const component = shallow(<TranslationCard {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('renders with the correct default values in input fields', () => {
        const component = mount(<TranslationCard {...props} />);
        const nativeInputDefaultValue = component
            .find('textarea.m-translation-card__native-field-input')
            .props()
            .value;

        const translatedInputDefaultValue = component
            .find('textarea.m-translation-card__translated-field-input')
            .props()
            .value;
            
        expect(nativeInputDefaultValue).toEqual(translation.native_word);
        expect(translatedInputDefaultValue).toEqual(translation.translated_word);
    });

    it('onTranslationFieldTextChange is called on translation field input change', () => {
        const component = mount(<TranslationCard {...props} />);
        const newValue = "Some new value";

        expect(component
            .find('textarea.m-translation-card__translated-field-input')
            .props()
            .value
        ).toEqual(translation.translated_word);   

        component
            .find('textarea.m-translation-card__translated-field-input')
            .simulate('change', { target: { value: newValue } });

        expect(component
            .find('textarea.m-translation-card__translated-field-input')
            .props()
            .value
        ).toEqual(newValue);
    });

    it('onNativeFieldTextChange is called on native field input change', () => {
        const component = mount(<TranslationCard {...props} />);
        const newValue = "Some new value";

        expect(component
            .find('textarea.m-translation-card__native-field-input')
            .props()
            .value
        ).toEqual(translation.native_word);
        
        component
            .find('textarea.m-translation-card__native-field-input')
            .simulate('change', { target: { value: newValue } });

        expect(component
            .find('textarea.m-translation-card__native-field-input')
            .props()
            .value
        ).toEqual(newValue);
    });
});