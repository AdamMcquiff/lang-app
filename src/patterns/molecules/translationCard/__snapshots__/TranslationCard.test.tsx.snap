// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationCard renders without crashing with all props 1`] = `
<div
  className="m-translation-card"
>
  <Button
    a11yLabel="Delete translation of Hello"
    className=""
    onClick={[Function]}
    textLabel="Delete"
  />
  <label
    className="m-translation-card__native-field-label"
  >
    <textarea
      className="m-translation-card__native-field-input"
      onChange={[Function]}
      placeholder=" translation"
      value="Hello"
    />
  </label>
  <hr />
  <label
    className="m-translation-card__translated-field-label"
  >
    <textarea
      className="m-translation-card__translated-field-input"
      onChange={[Function]}
      placeholder=" translation"
      value="Hallo"
    />
  </label>
</div>
`;
